### ISSUE TITLE
Out of memory issue W10

### ISSUE THREAD

**Issue by AmazingGabriel16** [yairm210/Unciv#7946]
**Platform**
Operating system type, major version, architecture (64/32 bit).
Windows 10 Pro Build 19044 64-bit

**Version**
Where you got the game and the exact version number.
4.2.14 (Build 758)

**Describe the bug**
Game sometimes running out of memory despite not using more than 2GB of RAM on the system. This results in a crash where you would have to close the app and restart it, but also an issue where it said that the phone did not have enough RAM.

**To Reproduce**
Play the game normally and you will encounter this issue in between turns

Commonly will occur when an enemy is on the screen

If game data is required, please add it here. From the menu, choose save game (or map), copy to clipboard, and paste here between the backtick blocks (or, if no such save is required, you may delete this):
<details><summary>Saved game</summary>

[Rome mega game.txt](https://github.com/yairm210/Unciv/files/9846861/Rome.mega.game.txt)

</details>

**Expected behavior**
For the game to use a whole ton more RAM instead of below 2GB, and therefore be able to successfully complete the turn.

**Screenshots**
![image](https://user-images.githubusercontent.com/23047921/197399780-85d10cb5-2380-4567-83b6-ad6fbb932f20.png)

**Additional context**
System specs:
Ryzen 9 5950X
8x4 3200 Mhz RAM (At the time, there was around 19GB being used, the game included as I was running background tasks. RAM usage did not increase much)

Used mods:
![image](https://user-images.githubusercontent.com/23047921/197400229-17d4dbb5-977c-441f-a281-c7658ce00993.png)

**Comment by greatalazar** [comment]
It is java thing, there is a maximum memory limit on JVM, you can change it in `GameSettings.json` file.

change it like this 
-Xmx\<amount of memory\> => -Xmx4G

this is a content of my `GameSettings.json` file

```json
{
  "jrePath": "jre",
  "classPath": [
    "Unciv.jar"
  ],
  "mainClass": "com.unciv.app.desktop.DesktopLauncher",
  "useZgcIfSupportedOs": false,
  "vmArgs": [
    "-Xmx6G",
    "-"
  ]
}
```

**Comment by yairm210** [comment]
Resolved in #7876 
Will roll out in 4.2.15
